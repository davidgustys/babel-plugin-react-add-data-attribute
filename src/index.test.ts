import pluginTester from 'babel-plugin-tester'
import path from 'path'
import plugin from '.'

pluginTester({
  plugin,
  babelOptions: { parserOpts: { plugins: ['jsx'] } },
  snapshot: true,
  tests: [
    {
      title: 'Functional Component',
      fixture: path.join(__dirname, '__fixtures__', 'functionComponent.tsx'),
    },
    {
      title: 'Arrow Function Component',
      fixture: path.join(__dirname, '__fixtures__', 'arrowFunction.tsx'),
    },
  ],
})
