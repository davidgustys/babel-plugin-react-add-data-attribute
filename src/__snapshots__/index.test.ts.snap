// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin Arrow Function Component: Arrow Function Component 1`] = `

const Div2 = () => <div />

const DivBody = () => {
  return <div />
}

      ↓ ↓ ↓ ↓ ↓ ↓

const Div2 = () => <div data-test="__fixtures___arrowFunction_div" />

const DivBody = () => {
  return <div data-test="__fixtures___arrowFunction_div" />
}


`;

exports[`unknown plugin Functional Component: Functional Component 1`] = `

function Div() {
  return <div />
}

function Nested() {
  return (
    <div>
      hello
      <div>world</div>
    </div>
  )
}

      ↓ ↓ ↓ ↓ ↓ ↓

function Div() {
  return <div data-test="__fixtures___functionComponent_div" />
}

function Nested() {
  return (
    <div data-test="__fixtures___functionComponent_div">
      hello
      <div data-test="__fixtures___functionComponent_div">world</div>
    </div>
  )
}


`;
